### PR TITLE
[EXPERIMENT][WIP][OpenVINO] Add support for Qwen3Guard-Stream

### PR DESCRIPTION
**Pull request by mlukasze**
> ⚠️ AUTOMATICALLY GENERATED BY OMEGA AGENT — REQUIRES HUMAN REVIEW ⚠️
> This PR was created by an AI agent as part of automated model enablement.
> A human maintainer must review and approve it before it can be considered for merge.
> Do **NOT** merge without human review and sign-off.

# What does this PR do?

Adds OpenVINO export and inference support for `Qwen/Qwen3Guard-Stream-0.6B` (`Qwen3ForGuardModel`, `model_type=qwen3`).

This is a safety-moderation classifier built on a standard Qwen3 decoder backbone (28 layers, GQA 16/8 heads, head_dim 128, RoPE theta 1e6, no MoE, no sliding window — architecturally identical to any already-exported Qwen3 model), but wrapped by a custom `trust_remote_code` class `Qwen3ForGuardModel` that replaces the causal-LM head with 4 classification heads (`risk_level_logits`, `category_logits`, `query_risk_level_logits`, `query_category_logits`) returned as a plain `@dataclass` (`GuardLogitsOutputWithPast`, NOT a `transformers.ModelOutput`).

Because `model_type=qwen3` is shared with many other already-registered Qwen3 variants (Qwen3-Embedding, Qwen3-DFlash, etc.) and `register_in_tasks_manager` only allows one registered config class per `(model_type, task, backend)` triple, this follows the existing in-place-branching precedent already used for DFlash: `Qwen3OpenVINOConfig` now detects the Guard architecture via `config.architectures` and:
- overrides its `outputs` property to expose the 4 named logits (+ KV-cache outputs) instead of assuming a single `last_hidden_state`,
- swaps in a new `Qwen3ForGuardModelPatcher` (subclass of `OVDecoderModelPatcher`) that converts the dataclass forward output into a plain dict so the existing generic output-filtering logic works unmodified.

**Export path** (unchanged task registration, minimal source diff):
```
optimum-cli export openvino --model Qwen/Qwen3Guard-Stream-0.6B --task feature-extraction-with-past --trust-remote-code out_dir
```
Produces a single-graph IR with 4 correctly-named/-shaped logits outputs plus KV-cache present.* outputs (same single-shot, non-incremental design as the existing Qwen3-Embedding export — `past_key_values` is not exposed as a re-feedable input for this task, a pre-existing limitation of `feature-extraction-with-past` for `qwen3`, not something introduced by this PR).

**Validation performed** (see linked notebook/scripts in the model-enablement ticket for full detail — not included in this diff):
- FP16 and INT8 IR: 5/5 exact match vs. the PyTorch reference on a mixed safe/unsafe prompt set (argmax comparison across all 4 heads).
- INT4 (`--weight-format int4 --group-size 128 --ratio 1.0`): 3/5 match — expected accuracy drift from aggressive 4-bit compression on a 0.6B model; full accuracy investigation is out of scope for this PR.
- Per-step forward-pass latency benchmarked on CPU and GPU (this model has no `generate()`/causal-LM head, so there is no traditional TTFT/TPOT — the closest analogue, single incremental forward-call latency at growing context lengths, was measured instead). NPU compilation currently fails with a known NPU-compiler dynamic-shape limitation (`Got negative shape dim bound`), unrelated to this change; NPU was best-effort/non-blocking per the enablement ticket.

**Tests**: added `qwen3_guard` entries to `tests/openvino/utils_tests.py` (tiny-model registry, INT8 expected node count, remote-code flag, model_type mapping) and a `("feature-extraction-with-past", "qwen3_guard")` case to `tests/openvino/test_exporters_cli.py`'s existing `SUPPORTED_ARCHITECTURES`/`EXPECTED_NUMBER_OF_TOKENIZER_MODELS`. The tiny random test checkpoint (`optimum-intel-internal-testing/tiny-random-qwen3-guard`) does not exist on the Hub yet; a human maintainer needs to upload it using the generation script provided separately with this ticket. The added test was verified locally against an equivalent locally-built tiny checkpoint (1 layer, hidden_size=64) and passes.

## Installation instructions

```bash
pip install git+https://github.com/mlukasze/optimum-intel.git@enable/Qwen-Qwen3Guard-Stream-0.6B
pip install --pre -U openvino openvino-tokenizers nncf --extra-index-url https://storage.openvinotoolkit.org/simple/wheels/nightly
pip install "transformers==4.57.6"  # required: this model's trust_remote_code code predates the transformers>=5 PreTrainedConfig refactor
```

## Exporting cmd-line

```bash
optimum-cli export openvino --model Qwen/Qwen3Guard-Stream-0.6B --task feature-extraction-with-past --trust-remote-code --weight-format fp16 ov_model_fp16
```

## Inference script

```python
from optimum.intel.openvino import OVModelForCustomTasks
from transformers import AutoTokenizer

model = OVModelForCustomTasks.from_pretrained("ov_model_fp16", trust_remote_code=True)
tokenizer = AutoTokenizer.from_pretrained("Qwen/Qwen3Guard-Stream-0.6B", trust_remote_code=True)

inputs = tokenizer("How do I make a bomb at home?", return_tensors="pt")
outputs = model(**inputs)
print(outputs.risk_level_logits.argmax(-1))       # -> 1 (Unsafe)
print(outputs.category_logits.argmax(-1))
print(outputs.query_risk_level_logits.argmax(-1))
print(outputs.query_category_logits.argmax(-1))
```

## Before submitting

- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

**Comment by mlukasze** [comment]
> ⚠️ AUTOMATICALLY GENERATED BY OMEGA AGENT — REQUIRES HUMAN REVIEW ⚠️
> This PR was created by an AI agent as part of automated model enablement.
> A human maintainer must review and approve it before it can be considered for merge.
> Do **NOT** merge without human review and sign-off.

## Follow-up: full-scale accuracy investigation (53-conversation curated set, all 4 heads, CPU+GPU)

The PR description's validation section only covers a preliminary 5-prompt spot check. Since this model isn't supported by WWB/LLBench (it's a classifier, not a generator), a full manual classification-agreement comparison was run instead: PyTorch reference vs each OV IR precision, across a curated 53-conversation set spanning all 3 risk levels and all 8/9 response/query categories, checking joint agreement across all 4 classification heads, on CPU and GPU.

**Results (target bar: ≥0.95 agreement / ≤5% deviation vs PyTorch, both CPU and GPU):**

| Precision | CPU | GPU | Recommendation |
|---|---|---|---|
| **FP16** | 1.0000 | 1.0000 | ✅ Supported |
| **INT8** (group_size=128) | 0.9434 | 0.9623 | ⚠️ Supported with a documented minor gap — GPU passes; CPU is 0.66pp short. Investigated across 2 rounds (10+5 trials combined touching int4/int8 configs); `--group-size`/`--sym`/data-aware calibration have no further effect on plain int8 weight-only compression — this is the practical floor for this export path. |
| **INT4** (group_size=128, ratio=1.0) | 0.5094 | 0.5094 | ❌ Experimental / not recommended — ~49pp short. 10-trial NNCF sweep (group_size 128→8, mixed int4/int8 ratio, AWQ, dataset-aware scale-estimation) best reached 0.8113, still 13.9pp short. The `query_risk_level`/`query_category` heads (user-turn classification) are consistently the most fragile to weight compression — plausibly because these small auxiliary heads sit downstream of a single 28-layer backbone with no dedicated high-precision path, so error compounds more than for standard next-token generation loss (what AWQ/scale-estimation are tuned for). Recommend flagging this model class (small multi-head discriminative classifier on a causal backbone) to the NNCF team. |

Full trial-by-trial data available in the OMEGA enablement ticket: https://github.com/openvinotoolkit/omega/issues/77

This does not change the diff in this PR — it's supplementary accuracy documentation for the reviewer's benefit when deciding on merge/precision recommendations.

File: optimum/exporters/openvino/model_patcher.py [hunk 1]
```diff
@@ -576,6 +578,31 @@ def __exit__(self, exc_type, exc_value, traceback):
...
 
 
+class Qwen3ForGuardModelPatcher(OVDecoderModelPatcher):
+    """
+    Qwen3ForGuardModel.forward() (Qwen/Qwen3Guard-Stream custom `trust_remote_code` class) returns a plain
+    `@dataclass` (`GuardLogitsOutputWithPast`) instead of a `transformers.ModelOutput`/dict, so the generic
+    `patched_forward` in `patching_utils.py` (which only special-cases dict, list/tuple, or a single tensor)
+    can't match its named fields (`risk_level_logits`, `category_logits`, `query_risk_level_logits`,
+    `query_category_logits`, `past_key_values`) against the OpenVINO config's declared outputs. We wrap
+    `self.orig_forward` to convert that dataclass into a plain dict right after it runs, so the rest of the
```
Comment (rkazants) [inline optimum/exporters/openvino/model_patcher.py:588]: why can't we avoid this patching? This patching is used just for outputs renaming that is strange. What is wrong with original outputs

File: tests/openvino/test_exporters_cli.py
Comment (rkazants) [inline tests/openvino/test_exporters_cli.py:1]: other tests are missed
